src/main/java/org/alfresco/rest/api/search/impl/ResultMapper.java

```diff
@@ -316,7 +316,7 @@ public SearchContext toSearchContext(SolrJSONResultSet solrResultSet, SearchRequ
                 for (Entry<String, Integer> fq:facetQueries.entrySet())
                 {
                     String filterQuery = null;
-                    if (searchQuery != null && searchQuery.getFacetQueries() != null)
+                    if (searchQuery.getFacetQueries() != null)
```

Comment:
Sonar was complaining here that if searchQuery would've been null, than it would raise a null pointer exception at line308: searchQuery.getFacetFormat(). So either the check is redundant because searchQuery cannot be null or it needs a fix prior to this check. I choosed a fix with minimal impact

Reply:
How do you know that an NPE is not actually the correct thing to do?
Skipping the if block when it's null could just mask problems and make things very difficult to debug.
Undo this change, please.

Reply:
So we get a NPE?  That is probably the correct thing.
Your fix will allow clients to pass in null without error, which is a change to the API.

Reply:
If searchQuery is arriving via an argument, then simply do an earlier check for null and throw IllegalArgumentException, which is the way to handle it, not to silently absorb problem arguments.

Reply:
Thanks for the comments. I've added a check for null